src/OpenApi/src/OpenApiSchemaGenerator.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using Microsoft.AspNetCore.Http;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.AspNetCore.OpenApi;
+
+internal static class OpenApiSchemaGenerator
+{
+    private static readonly Dictionary<Type, (string, string?)> simpleTypesAndFormats =
```

Comment:
`DateOnly` and `TimeOnly` get into tricky territory because of what is noted [here](https://github.com/dotnet/aspnetcore/issues/37045#issuecomment-1112664479).

`Half` and `Short` feel a little too niche for most web scenarios.

Timespan is a solid addition!